app-backend/batch/src/main/scala/geojsonImport/ImportGeojsonFiles.scala

```diff
@@ -0,0 +1,170 @@
+
+      opt[String]('u', "upload")
+        .required()
+        .action(
+          (u, conf) => {
+            conf.copy(upload = UUID.fromString(u))
+          }
+        )
+        .text("GeojsonUpload entry to process")
+    }
+}
+
+object ImportGeojsonFiles extends Config with LazyLogging {
+  val name = "import_geojson_files"
+
+  implicit val contextShift: ContextShift[IO] =
+    IO.contextShift(
+      ExecutionContext.fromExecutor(
+        Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder().setNameFormat("geojson-import-%d").build()
+        )
+      )
+    )
+  implicit val xa = RFTransactor.buildTransactor()
+
+  def processUploadToAnnotations(
+      upload: GeojsonUpload
+  ): List[Annotation.Create] = {
+    val s3Client = S3()
+    // download and parse files
+    upload.files.map { uri =>
+      {
+        logger.info(s"Downloading file: ${uri}")
+        // Only s3 uris are currently supported.
+        // TODO: hmm, this might be a security problem. Anyone can point RF at geojson file that we have
+        // access to, and this will let them view it even if they shouldn't be able to view it.
+        // I think we need a more formal system which requires signed uploads to specific directories, and
+        // reject any s3 urls that aren't in those directories like scene uploads
+        val s3Uri = new AmazonS3URI(URLDecoder.decode(uri, "utf-8"))
+        val s3Object = s3Client.getObject(s3Uri.getBucket, s3Uri.getKey)
+        val geojsonString =
+          new String(S3.getObjectBytes(s3Object), StandardCharsets.UTF_8)
+        logger.info("Annotations downloaded")
+        logger.info("Parsing annotations")
+        val annotations =
+          decode[AnnotationFeatureCollectionCreate](geojsonString) match {
+            case Right(fc) => fc.features.map(_.toAnnotationCreate).toList
+            case Left(e)   => throw e
+          }
+        logger.info("Annotations parsed")
+        annotations
+      }
+    }.flatten
+  }
+
+  def insertAnnotations(
+      annotations: List[Annotation.Create],
+      upload: GeojsonUpload
+  ): ConnectionIO[List[Annotation]] = {
+    // https://makk.es/blog/postgresql-parameter-limitation/
+    // max # of interpolated params in postgres driver = 32,767 (2 byte int)
+    // each annotation = 17 params
+    // 32,767 / 17 = 1927 annotations / batch. Cut that in half to 1000 to be safe.
+    for {
+      user <- UserDao.unsafeGetUserById(upload.createdBy)
+      inserted <- annotations
+        .grouped(1000)
+        .toList
+        .traverse(annotationBatch => {
+          val updatedAnnotationBatch =
+            annotationBatch
+              .map(_.copy(annotationGroup = Some(upload.annotationGroup)))
+          AnnotationDao.insertAnnotations(
+            updatedAnnotationBatch,
+            upload.projectId,
+            user,
+            Some(upload.projectLayerId)
+          )
+        })
+        .map { _.flatten }
```

Comment:
For huge annotation inputs (let's say a million annotations instead of 100k annotations), I think this flatten could be pretty costly -- since we only really care about the sizes being returned anyway, it might make sense to return the sizes instead of the annotation lists (`AnnotationDao.insertAnnotations(...) map { _.size }`) then sum up at the end, rather than create a giant array and then count its length.